### PR TITLE
#162006041 Fix room dropdown input field

### PR DESCRIPTION
**Pull request by matthewacha**
#### What does this PR do?
This PR enables the Room drop down field to be prefilled with `No beds found` when editing a multicity request with a trip that has no rooms added for the destination.

#### Description of Task to be completed?
- Autofill room field with `No beds found` for trips with no rooms while editing request

#### How should this be manually tested?
- Clone the repo - `git clone https://github.com/andela/travel_tool_front.git`
- Change into the project directory -` cd travel_tool_front`
- Pull and checkout into this branch - `bg-fix-bed-input-field-162006041 `
- Install project dependencies using `yarn install`
- Clone the [Travela backend](https://github.com/andela/travel_tool_back)
- Run database migrations using `yarn run db:rollmigrate`
- Check `.env.example` and set up your `.env` appropriately
- Start the Server and Frontend with `make start` and log into the application
- Create a new multi-city travel request with at least one trip to a destination with no beds/rooms
- Navigate to `Edit` the request

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[162006041](https://www.pivotaltracker.com/story/show/162006041)

#### Screenshots (if appropriate)
image

#### Questions:
N/A

### CODE REVIEW

**Comment by ssewilliam** [comment]
Nicely done @matthewacha

File: src/components/Forms/FormsAPI/Input/InputFields/DropdownSelect.jsx [hunk 1]
```diff
@@ -75,10 +75,11 @@ export default class DropdownSelect extends Component {
   };
 
   valueToChoiceLabel = (choices, value) => {
-    const choice =
+    let choice =
```
Comment (ezrqnkemboi) [inline src/components/Forms/FormsAPI/Input/InputFields/DropdownSelect.jsx]: Why are we using let here?

File: src/components/Forms/NewRequestForm/FormFieldsets/TravelDetailsItem.jsx [hunk 2]
```diff
@@ -55,23 +55,31 @@ class TravelDetailsItem extends Component {
...
     if (bedChoices.length < 1) {
-      bedChoices = [{ label: 'No beds found', value: -1 }];
+      this.setValues(values, itemId, 'No beds found', modalType);
+      bedChoices = [{ label: 'No beds found', value: 'No beds found'}];
     } else {
-      bedChoices = bedChoices.map(choice => ({
-        label: `${choice.rooms.roomName}, ${choice.bedName}`,
-        value: choice.id
-      }));
+      bedChoices = bedChoices.map(choice => {
```
Comment (ezrqnkemboi) [inline src/components/Forms/NewRequestForm/FormFieldsets/TravelDetailsItem.jsx]: I am able to get bed choice but unable to select one. The field also applies when there is no bed available.

File: src/components/Forms/FormsAPI/Input/InputFields/DropdownSelect.jsx (hunk 1 above)
Comment (matthewacha) [reply src/components/Forms/FormsAPI/Input/InputFields/DropdownSelect.jsx]: because we change its value on line 82

**Comment by ezrqnkemboi** [comment]
I am able to fill the bed space. The issue is that I get rooms No bed Space for a trip that had bed space and a blank field for the field with no beds. But, when I refresh the page, then edit, all the fields will have no bed.

File: src/components/Forms/NewRequestForm/FormFieldsets/TravelDetailsItem.jsx (hunk 2 above)
Comment (ezrqnkemboi) [reply src/components/Forms/NewRequestForm/FormFieldsets/TravelDetailsItem.jsx]: Working.

**Pull request merged**